modules/consensus/consensusset.go

```diff
@@ -131,6 +133,11 @@ func New(gateway modules.Gateway, persistDir string) (*ConsensusSet, error) {
 	}
 
 	go func() {
+		if cs.tg.Add() != nil {
+			return
+		}
+		defer cs.tg.Done()
```

Comment:
ditto; this means `Close` will not return until `cs.threadedInitialBlockchainDownload` has finished. That function should be modified to exit early (blocking for a maximum of one iteration) during shutdown. Specifically, it should check for `tg.IsStopped`, and it should `select` on `time.After` instead of sleeping.